app/src/main/java/com/spascoding/taskycourse/feature_agenda/presentation/event/EventViewModel.kt

```diff
@@ -0,0 +1,91 @@
+package com.spascoding.taskycourse.feature_agenda.presentation.event
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.spascoding.taskycourse.R
+import com.spascoding.taskycourse.core.data.local.UserInfoManager
+import com.spascoding.taskycourse.core.data.onError
+import com.spascoding.taskycourse.core.data.onSuccess
+import com.spascoding.taskycourse.core.presentation.UiText
+import com.spascoding.taskycourse.core.presentation.asUiText
+import com.spascoding.taskycourse.feature_agenda.domain.repository.AgendaRepository
+import com.spascoding.taskycourse.feature_auth.presentation.util.NameInitialsExtractor
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class EventViewModel @Inject constructor(
+    private var userInfoManager: UserInfoManager,
+    private val agendaRepository: AgendaRepository,
+) : ViewModel() {
+
+    var state = MutableStateFlow(EventViewModelState())
+        private set
+
+    init {
+        viewModelScope.launch {
+            userInfoManager.userInfoFlow.collect { userInfo ->
```

Comment:
Here, you'd only care about the latest emission, so I'd use `collectLatest`. `collect` is only useful if you really want to process every single emission, no matter if a new one already came in